hosts/tunnel.go

```diff
@@ -112,6 +112,23 @@ func getSSHConfig(username, sshPrivateKeyString string, useAgentAuth bool) (*ssh
 	if err != nil {
 		return config, err
 	}
+
+	if len(sshCertificateString) > 0 {
+		key, _, _, rest, err := ssh.ParseAuthorizedKey([]byte(sshCertificateString))
+		if err != nil || len(rest) > 0 {
+			return config, fmt.Errorf("Unable to parse SSH certificate: %v", err)
+		}
+
+		if cert, ok := key.(*ssh.Certificate); ok {
+			signer, err = ssh.NewCertSigner(cert, signer)
+			if err != nil {
+				return config, err
+			}
+		} else {
```

Comment:
i think this can be done with a simpler logic:
```
if _, ok := key.(*ssh.Certificate); !ok {
 return config, fmt.Errorf("Unable to cast public key to SSH Certificate")
}
signer, err = ssh.NewCertSigner(key.(*ssh.Certificate), signer)
if err != nil {
return config, err
}
```

Reply:
Agreed, made the change as recommended.